crates/store/re_grpc_client/src/address.rs

```diff
@@ -1,65 +1,111 @@
+use url::Url;
+
 /// The given url is not a valid Rerun storage node URL.
 #[derive(thiserror::Error, Debug)]
-#[error("URL {url:?} should follow rerun://addr:port/recording/12345")]
-pub struct InvalidAddressError {
-    url: String,
+#[error("URL {url:?} should follow rerun://addr:port/recording/12345 for recording or rerun://addr:port/catalog for catalog")]
+pub struct InvalidRedapAddress {
+    url: Url,
     msg: String,
 }
 
 /// Parsed `rerun://addr:port/recording/12345`
-pub struct Address {
-    pub addr_port: String,
+pub struct RecordingAddress {
```

Comment:
This could have become something like:
```
pub enum RedapAddress {
  Recording(...)
  Catalog
}
```

Reply:
I also thought about that, but the use cases are different and then you need matching in the consumer (unless I'm missing something obvious)